app/src/main/java/org/fossasia/openevent/general/aboutEvent/AboutEventActivity.kt

```diff
@@ -0,0 +1,130 @@
+
+    private val aboutEventViewModel by viewModel<AboutEventViewModel>()
+    private var id: Long = -1
+    private val EVENT_ID: String = "EVENT_ID"
+    private var isHideToolbarView: Boolean = false
+    private lateinit var eventExtra: Event
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.fragment_about_event)
+
+        setSupportActionBar(aboutEventToolbar)
+
+        if (supportActionBar != null) supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        aboutEventCollapsingLayout.title = " "
+        appbar.addOnOffsetChangedListener(this)
+
+        aboutEventViewModel.error.observe(this, Observer {
+            Toast.makeText(this, it, Toast.LENGTH_LONG).show()
+        })
+
+        val intent = intent
+        id = intent.getLongExtra(EVENT_ID, -1)
+
+        aboutEventViewModel.progressAboutEvent.observe(this, Observer {
+            it?.let { showProgressBar(it) }
+        })
+
+        aboutEventViewModel.loadEvent(id)
+
+        aboutEventViewModel.event.observe(this, Observer {
+            it?.let {
+                loadEvent(it)
+            }
+        })
+    }
+
+    private fun loadEvent(event: Event){
+        eventExtra = event
+        aboutEventContent.text = event.description
+        val dateString = StringBuilder("")
+
+        aboutEventDetails.text = dateString.append(getAboutEventFormattedDate(event.startsAt))
+                                           .append(" - ")
+                                           .append(getAboutEventFormattedDate(event.endsAt))
+
+        eventName.text = event.name
+    }
+
+    private fun showProgressBar(show: Boolean) {
+        progressBarAbout.isIndeterminate = show
+        progressBarAbout.visibility = if (show) View.VISIBLE else View.GONE
+    }
+
+    override fun onOffsetChanged(appBarLayout: AppBarLayout, verticalOffset: Int) {
+
+        val maxScroll = appBarLayout.totalScrollRange
+
+        Timber.d("MAX "+maxScroll)
+        val percentage =  Math.abs(verticalOffset).toFloat() / maxScroll.toFloat()
+
+        if (percentage == 1f && isHideToolbarView) {
+            //Collapsed
+            if (TextUtils.isEmpty(eventExtra.name)) {
+                detailsHeader.visibility = View.GONE
+                aboutEventCollapsingLayout.title = eventExtra.name
+                Timber.d("collapse "+eventExtra.name)
+                isHideToolbarView = !isHideToolbarView
+            } else {
+                detailsHeader.visibility = View.GONE
+                aboutEventCollapsingLayout.title = eventExtra.name
+                Timber.d("collapse 2"+eventExtra.name)
+                isHideToolbarView = !isHideToolbarView
+            }
+        } else if (percentage < 1f && !isHideToolbarView) {
+            //Not Collapsed
+            Timber.d("no collapse"+eventExtra.name)
+            detailsHeader.visibility = View.VISIBLE
+            aboutEventCollapsingLayout.title = " "
```

Comment:
@iamareebjamal I am facing problem in identifying the place where it can be used in xml actually.

Reply:
*I am changing the visibility of the detailsHeader 
*Changing the title of collapsing layout 
These are all based on the toolbar view visibility

Reply:
These all are based on collapsing toolbar's state AFAIK which can at least be used to change the title of itself

Reply:
We can just set the way of viewing of collapsing toolbar on scrolling in xml, by attaching scroll flags, I couldn't find a way to set a title for the layout for each state in xml

Reply:
Okay I think i found it :+1:

Reply:
@iamareebjamal I have seemingly found a solution where you just set the title, and that offset listener is not even used. But like both the event name and date are on a same line

Reply:
Actually just for the correct order of showing the details -> event name plus date, we need the offset listener. The title as you said will only have the event name, but we are setting the event name programitically itself

Reply:
There is expandedTitle and collapsedTitle option in XML

Reply:
That's what even I thought, but there isn't actually

Reply:
https://stackoverflow.com/questions/31662416/show-collapsingtoolbarlayout-title-only-when-collapsed Second solution

Reply:
Isn't that for just changing the text appearance ?

Reply:
Okay I think I understood :+1:

Reply:
That is a very buggy solution, it doesn't even work for some APIs and title still displays :(

Reply:
It doesn't work with the latest support libraries

Reply:
@iamareebjamal Programmitically is the best way to avoid bugs I think :+1:  What are your views ?

Reply:
OK if you say so, but programmatic methods are the most buggy solutions. They are mostly used for non-standard solutions, can cause memory leaks and increase the development surface area, meaning we have more code to maintain, debug and fix. If there is no solution in your opinion, we may move forward